doc/changelog.md

```diff
@@ -63,104 +108,96 @@
+  has added to Catalyst.
   [(#331)](https://github.com/PennyLaneAI/catalyst/pull/331)
 
-  For any compiler packages seeking to be registered in PennyLane, this PR adds the `entry_points`
-  metadata under the the group name `pennylane.compilers`, with the following entry points:
+  For any compiler packages seeking to be registered in PennyLane, the `entry_points`
+  metadata under the the group name `pennylane.compilers` must be added, with the following entry points:
 
   - `context`: Path to the compilation evaluation context manager. This context manager should have
-    the method context.is_tracing(), which returns True if called within a program that is being
+    the method `context.is_tracing()`, which returns True if called within a program that is being
     traced or captured.
 
   - `ops`: Path to the compiler operations module. This operations module may contain compiler
     specific versions of PennyLane operations. Within a JIT context, PennyLane operations may
     dispatch to these.
 
   - `qjit`: Path to the JIT compiler decorator provided by the compiler. This decorator should have
-    the signature `qjit(fn, *args, **kwargs)`, where fn is the function to be compiled.
+    the signature `qjit(fn, *args, **kwargs)`, where `fn` is the function to be compiled.
+* The compiler driver diagnostic output has been improved, and now includes failing IR as well as
+  names of failing passes.
+  [(#349)](https://github.com/PennyLaneAI/catalyst/pull/349)
 
-* Include the "Catalyst" utility dialect in our MLIR C-API.
-  [(#345)](https://github.com/PennyLaneAI/catalyst/pull/345)
+* The scatter operation in the Catalyst dialect now uses a SCF for loop to avoid ballooning
+  the compiled code.
+  [(#307)](https://github.com/PennyLaneAI/catalyst/pull/307)
+
+* The `CopyGlobalMemRefPass` pass of our MLIR processing pipeline now supports
+  dynamically shaped arrays.
+  [(#348)](https://github.com/PennyLaneAI/catalyst/pull/348)
 
 <h3>Breaking changes</h3>
 
 * The axis ordering for `catalyst.jacobian` is updated to match `jax.jacobian`. Assume we have
-  parameters of shape `[a,b]` and results of shape `[c,d]`. The jacobian would get the shape
-  `[c,d,a,b]` instead of `[a,b,c,d]`.
+  parameters of shape `[a,b]` and results of shape `[c,d]`, the returned Jacobian will now have
+  shape `[c, d, a, b]` instead of `[a, b, c, d]`.
   [(#283)](https://github.com/PennyLaneAI/catalyst/pull/283)
 
 <h3>Bug fixes</h3>
 
-* Enable AutoGraph to convert functions even when they are invoked through functional wrappers such
-  as `adjoint`, `ctrl`, `grad`, `jacobian`, etc.
-  [(#336)](https://github.com/PennyLaneAI/catalyst/pull/336)
-
-  The following should now succeed:
+* Fixes the issue with missing `CFP_t` in `StateVectorLQubitDynamic` when building against the
+  master branch of PennyLane-Lightning. This issue was introduced in [(#499)](https://github.com/PennyLaneAI/pennylane-lightning/pull/499).
```

Comment:
```suggestion
* An upstream change in the PennyLane-Lightning project was addressed to prevent compilation issues
  in the `StateVectorLQubitDynamic` class in the runtime.
  The issue was introduced in [#499](https://github.com/PennyLaneAI/pennylane-lightning/pull/499).
```

Reply:
Thank you! This is much clearer